File: json/json_bench.ts
const data = Array.from({ length: 10_000 }, (_, i) => ({
  id: i + 1,
  name: `user_${i + 1}`,
  active: (i + 1) % 2 === 0,
}));

const start = performance.now();

for (let i = 0; i < 100; i++) {
  const jsonStr = JSON.stringify(data);
  JSON.parse(jsonStr);
}

const elapsed = (performance.now() - start) / 1000;

console.log(`Time: ${elapsed.toFixed(3)}s`);

File: json/json_bench.rb
require 'json'

data = (1..10_000).map { |i| { id: i, name: "user_#{i}", active: i.even? } }

start = Process.clock_gettime(Process::CLOCK_MONOTONIC)

100.times do
  json_str = JSON.generate(data)
  JSON.parse(json_str)
end

elapsed = Process.clock_gettime(Process::CLOCK_MONOTONIC) - start

puts "Time: #{'%.3f' % elapsed}s"
